src/core/thread/mle.cpp

```diff
@@ -2747,19 +2746,6 @@ ThreadError Mle::HandleChildIdResponse(const Message &aMessage, const Ip6::Messa
         (mDeviceMode & ModeTlv::kModeFFD))
     {
         SuccessOrExit(error = mNetif.GetMle().ProcessRouteTlv(route));
-
-        for (uint8_t i = 0; i <= kMaxRouterId; i++)
-        {
-            if (route.IsRouterIdSet(i))
-            {
-                numRouters++;
-            }
-        }
-
-        if (mRouterSelectionJitterTimeout == 0 && numRouters < mNetif.GetMle().GetRouterUpgradeThreshold())
-        {
-            mRouterSelectionJitterTimeout = (otPlatRandomGet() % mRouterSelectionJitter) + 1;
-        }
```

Comment:
this lines are removed because `mRouterSelectionJitterTimeout` would be assigned in `HandleChildStart()` when `SetStateChild()`